controllers/machineset_controller.go

```diff
@@ -203,6 +209,12 @@ func (r *MachineSetReconciler) reconcile(ctx context.Context, cluster *clusterv1
 		return ctrl.Result{}, errors.Wrap(err, "failed to list machines")
 	}
 
+	if remediated, err := r.MachineHealthCheck.Remediate(ctx, allMachines.Items); err != nil {
+		return ctrl.Result{}, errors.Wrap(err, "failed to remediate machines")
+	} else if remediated {
+		return ctrl.Result{Requeue: true}, nil
```

Comment:
Probably worth a comment explaining why we are requeueing here

Reply:
Is immediate requeueing the expected goal here, or should this use RequeueAfter?

It seems like the next requeue will likely return the same list of machines with the `r.Client.List` call on line 205 that we would have if we just continued on here (assuming we are reflecting changes made during remediation to the resources to allMachines).

Reply:
Yeah, I didn't know that `Requeue: true` was immediate, I thought it just meant after some normal interval we requeue.

Reply:
It depends...  I believe it is subject to the exponential backoff, but assuming an initial reconciliation for the resource in the queue, the first requeue would effectively be immediate.

Reply:
Do we actually need to force-requeue vs just return? Wouldn't we get a notification for the deletion timestamp being set on the Machine in the MS anyways?